packages/context/src/binding-inspector.ts

```diff
@@ -215,10 +220,16 @@ export function createBindingFromClass<T = unknown>(
   cls: Constructor<T | Provider<T>>,
   options: BindingFromClassOptions = {},
 ): Binding<T> {
+  debug('create binding from class %s with options', cls.name, options);
   const templateFn = bindingTemplateFor(cls);
   let key = options.key;
   if (!key) {
-    key = buildBindingKey(cls, options);
+    try {
+      key = buildBindingKey(cls, options);
+    } catch (err) {
+      err.message += ` (while building binding key for class ${cls.name})`;
+      throw err;
```

Comment:
Can we move the `try-catch` to `createBindingFromClass` so that we can have the `class` information for all errors?

Reply:
This `try-catch` is already inside `createBindingFromClass` function.

```ts
export function createBindingFromClass<T = unknown>(
  cls: Constructor<T | Provider<T>>,
  options: BindingFromClassOptions = {},
): Binding<T> {
  debug('create binding from class %s with options', cls.name, options);
  const templateFn = bindingTemplateFor(cls);
  let key = options.key;
  if (!key) {
    try {
      key = buildBindingKey(cls, options);
    } catch (err) {
      err.message += ` (while building binding key for class ${cls.name})`;
      throw err;
    }
  }

  // ...
}
```

Are you suggesting to wrap the entire body of `createBindingFromClass` into a try-catch blog?

Reply:
Yes. As the method knows the class.

Reply:
Done.

While making this change, I did a small refactoring of `createBindingFromClass` and extracted a new helper function `applyClassBindingOptions`. This makes the patch smaller (less lines of code are changing the indentation level).